MaterialLib/FractureModels/FractureModelBase.h

```diff
@@ -27,6 +27,38 @@ template <int DisplacementDim>
 class FractureModelBase
 {
 public:
+    /// The MaterialStateVariables may store material model specific state
+    /// (other than sigma and eps), which are usually material history
+    /// dependent. The objects are stored by the user (usually in assembly per
+    /// integration point) and are created via \ref
+    /// createMaterialStateVariables().
+    struct MaterialStateVariables
+    {
+        virtual ~MaterialStateVariables() = default;
+        virtual void pushBackState() = 0;
+
+        void reset()
+        {
+            _is_tensile_stress = false;
+            _shear_yield_function = 0.0;
+        }
+
+        void setShearYieldFucntion(double Fs) { _shear_yield_function = Fs; }
+        double getShearYieldFucntion() const { return _shear_yield_function; }
+
+        void isTensileStress(bool flag) { _is_tensile_stress = flag; }
```

Comment:
Using this method the flag is set. For this reason I would prefer a method name `setTensileStress`.